sdk/src/main/java/com/google/cloud/dataflow/sdk/options/GcpOptions.java

```diff
@@ -183,24 +184,67 @@ void setCredentialFactoryClass(
   public static class DefaultProjectFactory implements DefaultValueFactory<String> {
     private static final Logger LOG = LoggerFactory.getLogger(DefaultProjectFactory.class);
 
+    private static String getUserHome() {
+      return System.getProperty("user.home");
+    }
+
+    private String getActiveConfig() throws IOException {
+      final String defaultConfigName = "default";
+
+      // get active configuration
+      File activeConfFile = new File(
+          getUserHome(),
+          ".config/gcloud/active_config");
```

Comment:
ping 2x @lukecwik - do you think we should wait for google-cloud-java or move forward?